# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/performance/PerformanceWorker.h
# Commit: 27c8e30bf969
# Full Hash: 27c8e30bf969e55fc01c5b97c1479244204531c5
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2022-01-25 16:37:35
# Regressor Bug: 1744025
# File Overlap Count: 1
# Description:
#   Bug 1744025: Use CheckedUnsafePtr<WorkerPrivate> in PerformanceWorker. r=dom-worker-reviewers,edenchuang
#   
#   Depends on D132797
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132799
# ==============================================================================

diff -r 38ebc27679a4 -r 27c8e30bf969 dom/performance/PerformanceWorker.h
--- a/dom/performance/PerformanceWorker.h	Tue Jan 25 08:53:02 2022 +0000
+++ b/dom/performance/PerformanceWorker.h	Tue Jan 25 08:53:02 2022 +0000
@@ -8,12 +8,11 @@
 #define mozilla_dom_PerformanceWorker_h
 
 #include "Performance.h"
+#include "mozilla/dom/WorkerPrivate.h"
 
 namespace mozilla {
 namespace dom {
 
-class WorkerPrivate;
-
 class PerformanceWorker final : public Performance {
  public:
   explicit PerformanceWorker(WorkerPrivate* aWorkerPrivate);
@@ -94,7 +93,7 @@
   }
 
  private:
-  WorkerPrivate* mWorkerPrivate;
+  CheckedUnsafePtr<WorkerPrivate> mWorkerPrivate;
 };
 
 }  // namespace dom
